.github/workflows/build.yml

```diff
@@ -111,37 +111,40 @@ jobs:
   validate:
     needs: [configure]
     runs-on: ubuntu-latest
-    name: Compile and Check (Merge Ref)
+    name: Compile and Check (Merge Ref) 🛠️
     steps:
-      - name: Env
+      - name: Env 📊
         run: printenv
         env:
           GITHUB_CONTEXT: ${{ toJson(github) }}
-      - name: Checkout code
+      - name: Checkout code 📥
         uses: actions/checkout@v5
         with:
           persist-credentials: false
-          ref: ${{ github.sha }}  # this is the default, just being explicit.
-      - name: Setup Python
+          ref: ${{ github.sha }} # this is the default, just being explicit.
+      - name: Setup Python 🐍
         uses: ./.github/actions/setup-python
-      - name: Setup Gradle
+      - name: Setup Gradle 🐘
         uses: ./.github/actions/setup-gradle
         with:
           java-version: 25
           gradle-cache-read-only: ${{ !inputs.is-trunk }}
           gradle-cache-write-only: ${{ inputs.is-trunk }}
           develocity-access-key: ${{ secrets.DEVELOCITY_ACCESS_KEY }}
-      - name: Compile and validate
+      - name: Compile and validate 🏛️
         env:
           SCAN_ARG: ${{ inputs.is-public-fork && '--no-scan' || '--scan' }}
         # Gradle flags
         # --build-cache:  Let Gradle restore the build cache
         # --info:         For now, we'll generate lots of logs while setting up the GH Actions
         # --scan:         Publish the build scan. This will only work on PRs from apache/kafka and trunk
         # --no-scan:      For public fork PRs, we won't attempt to publish the scan
-        run: |
-          ./gradlew --build-cache --info $SCAN_ARG check releaseTarGz -x test
-      - name: Archive check reports
+        run: ./gradlew --build-cache --info $SCAN_ARG check releaseTarGz -x test
+      - name: Spotless ✨
+        run: ./gradlew spotlessCheck --build-cache
```

Comment:
also wondering how spot is linked please, just as a question. Assuming its already covered somehow with check phase before. Thx.